core/tee/tee_svc_cryp.c

```diff
@@ -3487,30 +3487,33 @@ TEE_Result syscall_asymm_verify(unsigned long state,
 		goto out;
 	}
 
-	if (TEE_ALG_GET_MAIN_ALG(cs->algo) == TEE_MAIN_ALGO_ECDSA)
-		hash_algo = TEE_ALG_SHA1;
-	else
-		hash_algo = TEE_DIGEST_HASH_TO_ALGO(cs->algo);
-
-	res = tee_hash_get_digest_size(hash_algo, &hash_size);
-	if (res != TEE_SUCCESS)
-		goto out;
-
-	if (TEE_ALG_GET_MAIN_ALG(cs->algo) == TEE_MAIN_ALGO_DSA) {
+	if (TEE_ALG_GET_MAIN_ALG(cs->algo) != TEE_MAIN_ALGO_ECDSA) {
```

Comment:
I think this function would be easier to read if what's inside this block was copied/split into corresponding places for `case TEE_MAIN_ALGO_RSA:` and `case TEE_MAIN_ALGO_DSA:`. I think we can live with a duplicated call to `tee_hash_get_digest_size()`.

The reason it looks like it does today is because `TEE_MAIN_ALGO_ECDSA` was added afterwards.